src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -104,91 +104,65 @@ public ExamResource(UserRepository userRepository, CourseRepository courseReposi
-            return conflict();
+        if (exam.getCourse() == null || !exam.getCourse().getId().equals(courseId) || exam.getMaxPoints() <= 0) {
+            return badRequest();
         }
 
         if (exam.getVisibleDate() == null || exam.getStartDate() == null || exam.getEndDate() == null || !exam.getVisibleDate().isBefore(exam.getStartDate())
                 || !exam.getStartDate().isBefore(exam.getEndDate())) {
-            return conflict();
+            return badRequest();
         }
 
         // Check that exerciseGroups are not set to prevent manipulation of associated exerciseGroups
         if (!exam.getExerciseGroups().isEmpty()) {
-            return forbidden();
-        }
-
-        if (exam.getMaxPoints() <= 0) {
-            return conflict();
-        }
-
-        Optional<ResponseEntity<Exam>> courseAccessFailure = examAccessService.checkCourseAccessForInstructor(courseId);
-        if (courseAccessFailure.isPresent()) {
-            return courseAccessFailure.get();
+            return badRequest();
         }
 
-        Exam result = examRepository.save(exam);
-        return ResponseEntity.created(new URI("/api/courses/" + courseId + "/exams/" + result.getId()))
-                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, result.getTitle())).body(result);
+        Exam resultExam = examRepository.save(exam);
+        return ResponseEntity.created(new URI("/api/courses/" + courseId + "/exams/" + resultExam.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, resultExam.getTitle())).body(resultExam);
     }
 
     /**
-     * PUT /courses/{courseId}/exams : Updates an existing exam.
+     * PUT courses/{courseId}/exams/{examId} : Updates an existing exam.
      * This route does not save changes to the exercise groups. This should be done via the ExerciseGroupResource.
      *
      * @param courseId      the course to which the exam belongs
      * @param updatedExam   the exam to update
+     * @param examId        the id of the exam to be updated
      * @return the ResponseEntity with status 200 (OK) and with body the updated exam
-     * @throws URISyntaxException if the Location URI syntax is incorrect
      */
-    @PutMapping("/courses/{courseId}/exams")
+    @PutMapping("courses/{courseId}/exams/{examId}")
     @PreAuthorize("hasRole('INSTRUCTOR')")
-    public ResponseEntity<Exam> updateExam(@PathVariable Long courseId, @RequestBody Exam updatedExam) throws URISyntaxException {
+    public ResponseEntity<Exam> updateExam(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody Exam updatedExam) {
         log.debug("REST request to update an exam : {}", updatedExam);
-        if (updatedExam.getId() == null) {
-            return createExam(courseId, updatedExam);
+        if (!updatedExam.getId().equals(examId)) {
+            return badRequest("examId", "400", "ExamId in path and in Body don't match!");
```

Comment:
is not `conflict()` more suitable for this case? I'm just curious

Reply:
`conflict` is used when the state of the server and the request conflict with each other, however here the request path and the request body conflict with each other => `badrequest` imo

Reply:
ohh, okay.. bc, I had to check similar situation before, and I used conflict for that. I just wanted to know what is right response here.